src/NServiceBus.Core/Transports/MessageProcessingResult.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace NServiceBus.Transport
+{
+    /// <summary>
+    /// Controls how the transport should finalize the processing of the message.
+    /// </summary>
+    public class MessageProcessingResult
+    {
+        /// <summary>
+        /// Initializes the result.
+        /// </summary>
+        /// <param name="forceMessageRollback">Indicates that the message receive operation should be aborted and the message rolled back to the inputqueue.</param>
+        public MessageProcessingResult(bool forceMessageRollback)
+        {
+            ForceMessageRollback = forceMessageRollback;
+        }
+
+        /// <summary>
+        /// Indicates that the message receive operation should be aborted and the message rolled back to the inputqueue.
+        /// </summary>
+        public bool ForceMessageRollback { get; }
```

Comment:
I wonder if we should propagate the intent of the call to `AbortReceiveOperation` and call this `ReceiveOperationWasAborted`?

Reply:
I like it, will change

Reply:
Actually, when thinking some more about it I think `AbortReceiveOperation` is a better name since this is a "request" to the transport to abort the receive. The one actually doing the abort is still the transport so it hasn't happened yet at this stage?

Reply:
👍 `AbortReceiveOperation` works.